### PR TITLE
Add stubs for security factories

### PR DESCRIPTION
**Pull request by yguedidi**
ref symfony/symfony#49783

### CODE REVIEW

File: stubs/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AuthenticatorFactoryInterface.stub [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+use Symfony\Component\Config\Definition\Builder\NodeDefinition;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+interface AuthenticatorFactoryInterface
+{
+    /**
+     * @param array<string, mixed> $config
+     *
+     * @return string|string[]
```
Comment (ondrejmirtes) [inline stubs/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AuthenticatorFactoryInterface.stub:13]: When does it return a string and when does it return an array?

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: stubs/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AuthenticatorFactoryInterface.stub (hunk 1 above)
Comment (yguedidi) [reply stubs/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AuthenticatorFactoryInterface.stub:13]: It depends on the concrete implementation, for example:

- string: https://github.com/symfony/symfony/blob/6.3/src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AccessTokenFactory.php#L70
- array: https://github.com/symfony/symfony/blob/6.3/src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/CustomAuthenticatorFactory.php#L60

Comment (stof) [reply stubs/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AuthenticatorFactoryInterface.stub:13]: Returning a string is treated the same than returning an array containing a single item by the Symfony code calling this factory. It allows to simplify the most common case which only needs to create a single authenticator.

Comment (yguedidi) [reply stubs/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/AuthenticatorFactoryInterface.stub:13]: @ondrejmirtes is there anything else I can do?

**Review by stof — APPROVED** [review]

**Comment by ondrejmirtes** [comment]
Thank you!

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**